### PR TITLE
Run `pyupgrade --py38-plus` on the source code

### PR DESCRIPTION
**Pull request by sobolevn**
After `py3.7` support is gone (https://github.com/python/mypy/pull/15566), we can update our source code with new python syntax.

### CODE REVIEW

File: mypyc/ir/ops.py [hunk 1]
```diff
@@ -1575,5 +1574,5 @@ def visit_keep_alive(self, op: KeepAlive) -> T:
 # (Serialization and deserialization *will* be used for incremental
 # compilation but so far it is not hooked up to anything.)
 class DeserMaps(NamedTuple):
-    classes: Dict[str, "ClassIR"]
-    functions: Dict[str, "FuncIR"]
+    classes: dict[str, ClassIR]
```
Comment (AlexWaygood) [inline mypyc/ir/ops.py:1577]: PEP-563 forward references in NamedTuple classes make mypyc segfault IIRC

**Review by AlexWaygood — APPROVED** [review]
LGTM, assuming there are no more segfaults :-)

**Comment by ikonst** [comment]
Let's follow up by updating the blame ignored revisions list?

**Comment by sobolevn** [comment]
@ikonst it only touched imports, which (in my opinion) is not very important.
But, if it bothers you, why not :)

**Comment by ikonst** [comment]
I'm also seeing some f-strings and conversions to new union syntax. Not urgent, of course, can be done any time.

**Pull request merged**